Cairo Desktop/Cairo Desktop/CairoSettingsWindow.xaml.cs

```diff
@@ -392,5 +392,24 @@ private void btnChangeShell_Click(object sender, RoutedEventArgs e)
             if (LogoffChoice.HasValue && LogoffChoice.Value)
                 NativeMethods.Logoff();
         }
+
+        private void ChkRunAtLogOn_Click(object sender, RoutedEventArgs e)
+        {
+            RegistryKey key = Registry.CurrentUser.CreateSubKey(@"Software\Microsoft\Windows\CurrentVersion\Run");
+            var chkBox = (System.Windows.Controls.CheckBox)sender;
+
+            if (chkBox.IsChecked.Equals(false))
+            {
+                //Delete SubKey
+                key.DeleteValue("Cairo");
+            }
+            else
+            {
+                //Write SubKey
+                //Settings.RunAtLogOn = true;
+                key.SetValue("Cairo", @"C:\Program Files\Cairo Shell\CairoDesktop.exe");
```

Comment:
This makes an assumption that Cairo is installed in this location. It should use the path of the current running executable.